src/main/java/org/apache/datasketches/memory/internal/AllocateDirect.java

```diff
@@ -33,7 +33,7 @@
  * @author Lee Rhodes
  */
 @SuppressWarnings({"restriction","synthetic-access"})
-final class AllocateDirect implements AutoCloseable {
+final class AllocateDirect {
```

Comment:
Is it possible that the end user will forget to close the handle, now that `AllocateDirect` no longer implements `AutoCloseable`?

Reply:
This was redundant.  See [WritableDirectHandleImpl](https://github.com/apache/datasketches-memory/blob/AddExceptions/src/main/java/org/apache/datasketches/memory/internal/WritableDirectHandleImpl.java).

- The AllocateDirect class provides the means to allocate and close off-heap memory.  
- The WritableMemory provides the extensive API to access this memory once allocated.
- The WritableDirectHandleImpl inherits AutoCloseable from WritableHandle and joins the AllocateDirect and the WritableMemory (the API) together.

In this way the allocation of a resource, and the control of closing a resource are independent of each other and both are independent of the access API.  

Neat, huh :)

Reply:
... and the user accesses the ultimate public WritableMemory from the get() method of WritableHandle.  So the user doesn't have access to these internal classes.